### DISCUSSION TITLE
No path could be found for input?

### DISCUSSION THREAD

**Discussion by brad-4k** [valhalla/valhalla#3767] [General]
Thanks again for all your hard work!

I'm not sure if this is a "discussion" or an "issue" but I'll start here.

I have two points that my setup can't find a route between.  I've tried it on the valhalla1.openstreetmap.de server as well and got the same result.  When I try it on the valhalla.openstreetmap.de server, filling in the coordinates in the entry boxes, it does work.

What could I be doing wrong?

[https://valhalla1.openstreetmap.de/route?json={%22locations%22:[{%22lat%22:46.15329,%22lon%22:-60.22991},{%22lat%22:45.612822,%22lon%22:-61.3404}],%22costing%22:%22truck%22,%22directions_options%22:{%22units%22:%22miles%22}}](url)

[https://valhalla.openstreetmap.de/directions?profile=truck&wps=-60.22991%2C46.15329%2C-61.3404%2C45.612822](url)

**Comment by nilsnolde** [comment]
What is biting you here is a mismatch of the Valhalla defaults for trucks and the web app's default for trucks: https://github.com/gis-ops/valhalla-app/issues/31. 

So, the web app defaults to 1.9 m height and 1.6 m width, which is the default config for car. When you do the plain request without costing options, Valhalla has its own truck defaults of 4.1 m height. Somewhere along the route there must be a road which doesn't allow vehicles to be that high (bridge or so?), so it fails the route (even though 2 paths seem to lead from origin to destination).

I'm not sure if I like that Valhalla makes those assumptions per default on the truck profile. In Germany many things can be considered "trucks" (I'm sure in other countries too), and Valhalla's defaults are describing a pretty huge truck with 21 tons, 4 m height and a length of 21 m. IMHO it'd have been easier for everyone to not have any defaults on those dimensions and only consider them when they're explicitly set in the costing options. Especially since they can fail routes, i.e. it's hard exclusive properties. But it's always been like that and it doesn't seem to have hurt anyone so far, so I might be alone with my irritation :laughing:

**Comment by kevinkreiser** [comment]
we get all our truck tagging from the `hgv` style tags in osm which of course refer to https://en.wikipedia.org/wiki/Large_goods_vehicle

tldr the philosophy is there are a lot of kinds of trucks like there are a lot of kinds of bikes. to allow for all the variation we have the generic term truck and then we have to pick a default. we decided to go with the most common routing use-case which was 18 wheelers. this can be confusing in that it will fail some routes but its more useful than defaulting to a truck that is basically a car.

i do agree that this can be a gotcha though. we can put more in the way of docs... we could also return the costing options we ended up using for the request for more information to be returned to the caller.

**Reply by nilsnolde** [reply]
> we could also return the costing options we ended up using for the request for more information to be returned to the caller.

Sounds good to me!

**Comment by brad-4k** [comment]
These are wonderful explanations, thank you.  In these cases we have a backup option (hit another service to get the route) but I would prefer to stick to Valhalla in all cases.
As a thought, what if the cost model was a list (in order) of cost models to try?  Failure in one results in a try in the next.

**Reply by kevinkreiser** [reply]
rather than complicate the api, id rather the client retry with relaxed costing options. we can make changes in the future to the error return so the client can better estimate whether its worth a retry with different relaxed options.

**Reply by brad-4k** [reply]
Thanks Kevin.  I'll poke our other team to make that an option.

I'll also look into what it takes to make a new cost model, something matching a smaller "truck", that can be used in these cases.

**Reply by nilsnolde** [reply]
Is there smth specific missing in the costing options for truck that makes you think about a new model?

**Reply by brad-4k** [reply]
I need something like a "small truck" model (probably bad term) that would work on these smaller roads.

**Reply by nilsnolde** [reply]
You mean, being able to specify `weight`, `height`, `width`, `length`, `axle_load` & `hazmat` for every request?

**Reply by brad-4k** [reply]
I think a "light-truck" model would suffice.  Or "panel van" or whatever the proper nomenclature would be.

**Reply by nilsnolde** [reply]
We’d be happy to create a custom model for you (which would need to live in a fork, since there’s no use for the main project). You already have my email adress if you need support.

In the meantime you can use the [flexible costing options](https://github.com/valhalla/valhalla/blob/master/docs/api/turn-by-turn/api-reference.md#costing-options) 😉
